src/main/java/org/spongepowered/common/statistic/SpongeBlockStatistic.java

```diff
@@ -41,14 +43,22 @@
 public final class SpongeBlockStatistic extends StatCrafting implements BlockStatistic, TypedSpongeStatistic {
 
     private String spongeId;
+    private Translation translation;
 
     public SpongeBlockStatistic(String statId, String itemName, ITextComponent statName, Item item) {
         super(statId, itemName, statName, item);
+        String args;
+        if (statId.equals("stat.mineBlock.")) {
+            args = Block.getBlockFromItem(item).getLocalizedName();
+        } else {
+            args = new ItemStack(item).getDisplayName();
+        }
+        translation = new SpongeTranslation(statId.substring(0, statId.length() - 1), args);
```

Comment:
`this`
also why did you initialize this in the constructor while all the others are lazy initialized in the getter?

Reply:
Because I need the `statId` and the `item`